### PR TITLE
822 geometry of patient body

### PR DESCRIPTION
**Pull request by ostatni5**
Add CT:
image

Preview CT:
image

POC resultCT:
image

### CODE REVIEW

File: src/JsRoot/GraphData.tsx [hunk 1]
```diff
@@ -38,14 +39,19 @@ export interface Page0D extends IPage {
 	dimensions: 0;
 }
 
+export interface PageCT extends IPage {
+	dimensions: 'CT';
+	resultsUrl: string;
+}
```
Comment (Derstilon) [inline src/JsRoot/GraphData.tsx:45]: What does CT stand for? Is this data format dictated by the backend response? We could use some renames to clarify that Page types are distinguishable by dimension count or this 'CT' value.

File: src/ThreeEditor/components/Sidebar/propeteries/category/CTConfiguration.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+									'pathOnServer',
+									value
+								)
+							);
+						}}
+					/>
+				</>
+			)}
+		</PropertiesCategory>
+	);
```
Comment (Derstilon) [inline src/ThreeEditor/components/Sidebar/propeteries/category/CTConfiguration.tsx:37]: Maybe reduce indent by writing it like this?
```suggestion
	const handlePathOnServerChange = useCallback(
		(value: string) =>
			editor.execute(
				new SetValueCommand(editor, watchedObject.object, 'pathOnServer', value)
			),
		[watchedObject.object, editor]
	);

	return (
		<PropertiesCategory category='CT Configuration' visible={visibleFlag}>
			{visibleFlag && (
				<TextPropertyField
					label='Path on server'
					value={watchedObject.pathOnServer}
					onChange={handlePathOnServerChange}
				/>
			)}
		</PropertiesCategory>
	);
```

**Review by Derstilon — CHANGES_REQUESTED** [review]
We have a lot of files with a same construction
```html
<PropertiesCategory visible={visibleFlag}>
	{visibleFlag && (<Smth/>)}
</PropertiesCategory>
```
image

Is it possible to implement visibleFlag condition inside of Properties Category implementation?
Something like this:
```tsx
return (
	<ReguralContentsOfPropertiesCategory>
		{visible && (props.children)}
	</ReguralContentsOfPropertiesCategory>
)
```

File: src/JsRoot/GraphData.tsx (hunk 1 above)
Comment (ostatni5) [reply src/JsRoot/GraphData.tsx:45]: I'm not sure but I was to use name 'CT' when referring to patient objects like meshes etc.

**Comment by ostatni5** [comment]
> We have a lot of files with a same construction
> 
> ```
> <PropertiesCategory visible={visibleFlag}>
> 	{visibleFlag && (<Smth/>)}
> </PropertiesCategory>
> ```
> 
> image
> 
> Is it possible to implement visibleFlag condition inside of Properties Category implementation? Something like this:
> 
> ```tsx
> return (
> 	<ReguralContentsOfPropertiesCategory>
> 		{visible && (props.children)}
> 	</ReguralContentsOfPropertiesCategory>
> )
> ```

Well probably it is possible but  I remember there was some problems with such solution. Make for it another issue.

File: src/JsRoot/GraphData.tsx (hunk 1 above)
Comment (grzanka) [reply src/JsRoot/GraphData.tsx:45]: this is also puzzling me - what is stored in the dimesions ? At first glance this looks like: X,Y,Z or cm or mm.

File: src/JsRoot/GraphData.tsx [hunk 3]
```diff
@@ -38,14 +39,19 @@ export interface Page0D extends IPage {
 	dimensions: 0;
 }
 
+export interface PageCT extends IPage {
+	dimensions: 'CT';
```
Comment (grzanka) [inline src/JsRoot/GraphData.tsx:43]:
```suggestion
	dimensions: '3D_datafile_CT';
```

File: src/JsRoot/GraphData.tsx (hunk 1 above)
Comment (Derstilon) [reply src/JsRoot/GraphData.tsx]: It has int value representing number of dimensions for input. So usually it's 0, 1, 2 or 3.

File: src/ThreeEditor/components/Sidebar/propeteries/category/CTConfiguration.tsx (hunk 2 above)
Comment (Derstilon) [reply src/ThreeEditor/components/Sidebar/propeteries/category/CTConfiguration.tsx:37]: It's easier to see where component ends.

File: src/JsRoot/GraphData.tsx [hunk 4]
```diff
@@ -17,7 +18,7 @@ export type pageData = {
 interface IPage {
 	filterRef?: FilterJSON;
 	name?: string;
-	dimensions: number;
+	dimensions: number | string;
```
Comment (grzanka) [inline src/JsRoot/GraphData.tsx:21]: I was thinking of simplifying the design here. 
@ostatni5 what do you think of keeping the dimension as integer.
Then is dimensions == 3 then we present to the user results in form the same as `ResultsCT.tsx` ?
In fact results from CT scan are basically the same as of any 3-D scorer.

Comment (ostatni5) [reply src/JsRoot/GraphData.tsx:21]: So if the results are 3D will it always be a link?

Comment (grzanka) [reply src/JsRoot/GraphData.tsx:21]: Yes

**Review by grzanka — APPROVED** [review]
LGTM

File: src/WrapperApp/components/Results/ResultsCT.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import { Box, Button, Stack, Typography } from '@mui/material';
+import { PageCT } from '../../../JsRoot/GraphData';
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+import ContentCopyIcon from '@mui/icons-material/ContentCopy';
+
+export function ResultCT(props: { page: PageCT; title?: string }) {
```
Comment (grzanka) [inline src/WrapperApp/components/Results/ResultsCT.tsx:6]: Shouldn't be this renamed to Results3D ? Then all 3D results will be displayed here

File: src/JsRoot/GraphData.tsx [hunk 6]
```diff
@@ -38,14 +39,19 @@ export interface Page0D extends IPage {
...
+
 export type Estimator = {
 	name: string;
 	metadata?: unknown;
 	pages: Page[];
 	scoringOutputJsonRef?: ScoringOutputJSON;
 };
 
-export type Page = Page2D | Page1D | Page0D;
+export type Page = Page2D | Page1D | Page0D | PageCT;
```
Comment (grzanka) [inline src/JsRoot/GraphData.tsx:54]: Lets rename PageCT to Page3D and treat CT as regular 3D output

**Review by grzanka — CHANGES_REQUESTED** [review]
see comments inline

**Review by grzanka — APPROVED** [review]
ok

File: src/JsRoot/GraphData.tsx [hunk 7]
```diff
@@ -38,14 +39,19 @@ export interface Page0D extends IPage {
...
+
 export type Estimator = {
 	name: string;
 	metadata?: unknown;
 	pages: Page[];
 	scoringOutputJsonRef?: ScoringOutputJSON;
 };
 
-export type Page = Page2D | Page1D | Page0D;
+export type Page = Page2D | Page1D | Page0D | Page3D;
```
Comment (Derstilon) [inline src/JsRoot/GraphData.tsx:54]:
```suggestion
export type Page = Page3D | Page2D | Page1D | Page0D;
```

File: src/WrapperApp/components/Results/Results3D.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
+import { Box, Button, Stack, Typography } from '@mui/material';
+import { Page3D } from '../../../JsRoot/GraphData';
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+import ContentCopyIcon from '@mui/icons-material/ContentCopy';
+
+export function Result3D(props: { page: Page3D; title?: string }) {
+	const { page, title } = props;
+
+	return (
+		<Box sx={{ margin: '1rem 0' }}>
```
Comment (Derstilon) [inline src/WrapperApp/components/Results/Results3D.tsx:10]: use spacing value from theme to define margin
```suggestion
		<Box sx={{ margin: ({spacing}) => spacing(1, 0) }}>
```

File: src/WrapperApp/components/Results/Results3D.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+					justifyContent: 'flex-start'
+				}}>
+				<Stack direction={'row'} gap='1rem' alignItems={'center'}>
+					<Typography variant='h6'>Url on server: </Typography>
+					<Typography variant='h6' fontWeight={'bold'}>
+						{page.resultsUrl}
+					</Typography>
+					<CopyToClipboard text={page.resultsUrl}>
+						<Button variant='text' color='secondary'>
+							<ContentCopyIcon sx={{ marginRight: '0.5rem' }} /> Copy to clipboard
```
Comment (Derstilon) [inline src/WrapperApp/components/Results/Results3D.tsx:25]:
```suggestion
							<ContentCopyIcon sx={{ marginRight: ({spacing}) => spacing(.5) }} /> Copy to clipboard
```

**Review by Derstilon — APPROVED** [review]
image LGTM

**Pull request merged**